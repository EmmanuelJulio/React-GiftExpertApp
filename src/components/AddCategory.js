import React, { useState } from "react";
import PropType from 'prop-types'

export  const AddCategory =({setCategories})=> {
  const [inputValue, setinputValue] = useState('');
  const handleInputChange = (e) => {
    setinputValue(e.target.value);
    console.log("handle input change llamado")
  };
  const handleSubmint = (e) => {
      e.preventDefault()
            if(inputValue.trim().length>2){
                setCategories(cats=>[inputValue,...cats])
                setinputValue('')

            }
        
  };
  return (
    <form onSubmit={handleSubmint}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      ></input>
    </form>
  );
}

AddCategory.propTypes={
    setCategories: PropType.func.isRequired
}