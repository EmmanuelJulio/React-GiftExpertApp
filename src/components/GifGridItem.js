import React from 'react'

export const GifGridItem = ( {id,title,url} ) => {
  
    
    return (
        <div className="card animate__animated  animate__backInLeft">
         <img src={url} alt={title}/>
         <p>{title}</p>
        </div>
    )
}
/*
1. Enzyme
2. Enzyme to json
3. Debe de mostrar el componente correctamente
 *shallow
 *wrapper 
 *wrapper .toMatchSnaphop().


*/