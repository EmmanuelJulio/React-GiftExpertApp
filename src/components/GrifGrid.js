import React from "react";
import { Fragment } from "react/cjs/react.production.min";
import { useFetchGifs } from "../hooks/useFetchGif";
 import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';
export const GrifGrid = ({ category }) => {
  
    const {data:images,loading}= useFetchGifs( category );

  return (
    <Fragment>
    <h3 className="animate__animated  animate__fadeIn">{category}</h3>
   {loading&&<p className="animate__animated  animate__flash">Loading</p>}
    <div className="card-grid">
      
      
        {
          images.map(img=>(
            <GifGridItem
            key={img.id}
            {...img}
            />
          ))
        }
       
    </div>
    </Fragment>
  );
};
GrifGrid.propTypes={
  category:PropTypes.string.isRequired
}