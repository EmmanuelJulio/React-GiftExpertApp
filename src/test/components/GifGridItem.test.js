import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"
import React from 'react';

describe('Pruebas en GifGridItem', () => {
    const title ='Un titulo';
    const url ='https://blabla.com'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('Debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Deve de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.props('src'))
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    })
    test('Debe de tener animate__backInLeft', () => {
        const div = wrapper.find('div');
        const classname=div.prop('className');
        expect(classname.includes('animate__backInLeft')).toBe(true)
    })
    
    
})
