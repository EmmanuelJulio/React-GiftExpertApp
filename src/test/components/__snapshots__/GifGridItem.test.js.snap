// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated  animate__backInLeft"
>
  <img
    alt="Un titulo"
    src="https://blabla.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
